akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -3726,4 +3726,64 @@ trait FlowOpsMat[+Out, +Mat] extends FlowOps[Out, Mat] {
   def monitor: ReprMat[Out, (Mat, FlowMonitor[Out])] =
     monitorMat(Keep.both)
 
+  /**
+   * This is a more scalable and general case of [[groupedWeightedWithin]]
+   * which groups a stream into vectors based on custom weight and time.
+   * The problem with that solution is each grouped vector must fit into memory before emitting to the next stage.
+   * That won't work for the use case of writing large files.
+   * The desirable behavior is to write data as they come as opposed to accumulate everything until the group slicing condition is met.
+   * In this case, the output channel needs to be closed if there is no data arriving within certain time to avoid connection timeout.
+   * This custom flow uses custom aggregator to support such use cases.
+   * Upstream inputs are continuously aggregated as they arrive.
+   * The currently aggregated elements can be terminated and emitted based on custom conditions
```

Comment:
I think this needs some aligning with other operator API docs, suggestion:

```
Aggregate input elements into an arbitrary data structure that can be completed and emitted downstream when some condition is met or a timer triggers.

It can be thought of a more general [[groupedWeightedWithin]]. 

'''Emits when''' a the aggregation function decides the aggregate is complete or the timer function returns true

'''Backpressures when''' downstream backpressures and the aggregate is complete

'''Completes when''' upstream completes and the last aggregate has been emitted downstream

'''Cancels when''' downstream cancels
```